### DISCUSSION TITLE
Alembic path error

### DISCUSSION THREAD

**Discussion by DeLaRiva** [optuna/optuna#4314] [General]
I created an app ViPar with a GUI which makes use of Optuna. I'm quite happy with the entire project, especially, for sure, with Optuna which plays the major role.
To make it user friendly I designed a GUI and created an .exe file with PyInstaller. Now, when I want to run an optimization procedure from within the GUI I receive this error:

`alembic.util.exc.CommandError: Path doesn't exist: 'C:\\Users\\DWIEBW~1\\AppData\\Local\\Temp\\_MEI247442\\optuna\\storages\\_rdb\\alembic'.  Please use the 'init' command to create a new scripts folder.`

whereas everything works fine when I start the app as a python file itself from the terminal. So I guess something happens when PyInstaller is creating the .exe file but it's definitely over my head. I don't know how to track down the issue and not speaking of how to resolve it.

Through the app I receive more information which is:

```
Closed due to an error. This is the full error report:

Original exception was:
Traceback (most recent call last):
  File "ViPar.py", line 571, in clickme
  File "ViPar.py", line 752, in optimize
  File "optuna\_convert_positional_args.py", line 63, in converter_wrapper
  File "optuna\study\study.py", line 1147, in create_study
  File "optuna\storages\__init__.py", line 32, in get_storage
  File "optuna\storages\_rdb\storage.py", line 223, in __init__
  File "optuna\storages\_rdb\storage.py", line 1085, in __init__
  File "optuna\storages\_rdb\storage.py", line 1112, in _init_alembic
  File "optuna\storages\_rdb\storage.py", line 1167, in get_head_version
  File "optuna\storages\_rdb\storage.py", line 1209, in _create_alembic_script
  File "alembic\script\base.py", line 210, in from_config
  File "alembic\script\base.py", line 95, in __init__
alembic.util.exc.CommandError: Path doesn't exist: 'C:\\Users\\DWIEBW~1\\AppData\\Local\\Temp\\_MEI247442\\optuna\\storages\\_rdb\\alembic'.  Please use the 'init' command to create a new scripts folder.
```

Here is also a link to [stackoverflow ](https://stackoverflow.com/questions/75055387/created-exe-with-pyinstaller-path-doesnt-exist) where I raised this issue as well but I'm thinking that here is a higher chance that people here will pay attention to this and knowing something of it.
Thanks a lot in advance!

**Comment by DeLaRiva** [comment]
Ok, I figured that when I comment out the line 

`#storage_name = f"sqlite:///{study_name}.sqlite3"`
for
```
study = optuna.create_study(
                                    study_name=study_name
                                    #storage = storage_name
                                    , load_if_exists = False
                                    , directions = ["maximize"])
```
it works. Finally and happily.
But nevertheless, any idea how to resolve this? I would like to be able to store such files. What do I have to change to make this possible?

As mentioned above, when I run it from the terminal via `python ViPar.py` the GUI and everything works. Also the sqlite3 files are created. What is different when PyInstaller gets its hand on this?

**Comment by gen740** [comment]
Could you make a minimal reproduction code?

**Comment by RemiN1G** [comment]
I found the problem, when you use Pyinstaller, it'll create a python environment and run your program. The problem is located in Storage.py with the function create_alembic_config 
```
   def _create_alembic_config(self) -> "alembic_config.Config":
        alembic_dir = os.path.join(os.path.dirname(__file__), "alembic")
        config = alembic_config.Config(os.path.join(os.path.dirname(__file__), "alembic.ini"))
        config.set_main_option("script_location", escape_alembic_config_value(alembic_dir))
        config.set_main_option("sqlalchemy.url", escape_alembic_config_value(self.url))
        return config

```

In this function we are using the path of the current file 

> os.path.dirname(\_\_file__) 

but due to Pyinstaller there is no file at this location. I found a workaround by simply adding a folder optuna with storage folder and files inside to your application folder. For the case of a --Onefile pyinstaller i don't know how to fix that.

I hope that'll help you.
